extension/meminfo.c

```diff
@@ -36,15 +36,80 @@ zend_module_entry meminfo_module_entry = {
     STANDARD_MODULE_HEADER,
     "meminfo",
     meminfo_functions,
+    PHP_MINIT(meminfo),
+    PHP_MSHUTDOWN(meminfo),
     NULL,
     NULL,
+    PHP_MINFO(meminfo),
+    MEMINFO_VERSION,
+    PHP_MODULE_GLOBALS(meminfo),
+    PHP_GINIT(meminfo),
     NULL,
     NULL,
-    NULL,
-    MEMINFO_VERSION,
-    STANDARD_MODULE_PROPERTIES
+    STANDARD_MODULE_PROPERTIES_EX
 };
 
+PHP_GINIT_FUNCTION(meminfo)
+{
+	meminfo_globals->dump_on_limit = 0;
+}
+
+PHP_MINFO_FUNCTION(meminfo)
+{
+    DISPLAY_INI_ENTRIES();
+}
+
+#if   PHP_VERSION_ID < 70200 /* PHP 7.1 */
+static void meminfo_zend_error_cb(int type, const char* error_filename, const uint error_lineno, const char* format, va_list args)
+#elif PHP_VERSION_ID < 80000 /* PHP 7.2 - 7.4 */
+static void meminfo_zend_error_cb(int type, const char* error_filename, const uint32_t error_lineno, const char* format, va_list args)
+#elif PHP_VERSION_ID < 80100 /* PHP 8.0 */
+static void meminfo_zend_error_cb(int type, const char* error_filename, const uint32_t error_lineno, zend_string* message)
+#else                        /* PHP 8.1 */
+static void meminfo_zend_error_cb(int type, zend_string* error_filename, const uint32_t error_lineno, zend_string* message)
+#endif
+{
+#if PHP_VERSION_ID < 80000
+	const char* msg = format;
+#else
+	const char* msg = ZSTR_VAL(message);
+#endif
+
+	if (EXPECTED(!should_autodump(type, msg))) {
```

Comment:
I think this just looks goofy because of all the pre-processor conditions. These statements are inside of the function body of `static void meminfo_zend_error_cb(...)`. Unless I'm missing something else?

Reply:
Oops, I didn't notice there was a tabs/spaces issue with the indention. I've changed the tabs into spaces to match the rest of the file.